lib/kinesis_client/stream/coordinator.ex

```diff
@@ -116,12 +116,21 @@ defmodule KinesisClient.Stream.Coordinator do
 
         {:noreply, %{state | shard_graph: shard_graph, shard_ref_map: map}}
 
-      other ->
-        Logger.info("Stream is not in active state, sleeping... Stream state: #{inspect(other)}")
+      {:error, reason} ->
+        Logger.info("[kcl_ex] error describing stream shards with result #{inspect(reason)}")
+
+        if state.startup_attempt < @max_startup_attempts do
+          :timer.sleep(state.retry_timeout)
+          state = %{state | startup_attempt: state.startup_attempt + 1}
+          notify({:retrying_describe_stream, self()}, state)
+          {:noreply, state, {:continue, :describe_stream}}
```

Comment:
We'll try up to three times to describe the stream shards, before manually terminating the GenServer. We wait 30 seconds between each attempt (the default configuration).